File: frontend/src/services/questionService.js
import api from '@/utils/api'

const questionService = {
  publishQuestion(title, content, tags) {
    return api.post('/questions', null, {
      params: { title, content, tags }
    })
  },

  getQuestionDetail(id) {
    return api.get(`/questions/${id}`)
  },

  getQuestionList(sortBy = 'latest', tags = null, solved = null, pageNum = 1, pageSize = 20) {
    return api.get('/questions', {
      params: { sortBy, tags, solved, pageNum, pageSize }
    })
  },

  searchQuestions(keyword, pageNum = 1, pageSize = 20) {
    return api.get('/questions/search', {
      params: { keyword, pageNum, pageSize }
    })
  },

  getUserQuestions(userId, pageNum = 1, pageSize = 20) {
    return api.get(`/questions/user/${userId}`, {
      params: { pageNum, pageSize }
    })
  },

  updateQuestion(id, title, content, tags) {
    return api.put(`/questions/${id}`, null, {
      params: { title, content, tags }
    })
  },

  deleteQuestion(id) {
    return api.delete(`/questions/${id}`)
  },

  acceptAnswer(questionId, answerId) {
    return api.post(`/questions/${questionId}/accept/${answerId}`)
  }
}

export default questionService

File: backend/src/main/java/com/qa/service/QuestionService.java
package com.qa.service;

import com.qa.common.PageResult;
import com.qa.dto.QuestionDTO;
import com.qa.entity.Question;

/**
 * 问题业务服务
 */
public interface QuestionService {
    
    /**
     * 发布问题
     */
    Question publishQuestion(Long authorId, String title, String content, String tags);
    
    /**
     * 获取问题详情
     */
    QuestionDTO getQuestionDetail(Long questionId, Long userId);
    
    /**
     * 获取问题列表
     */
    PageResult<QuestionDTO> getQuestionList(String sortBy, String tags, Boolean solved, 
                                             Integer pageNum, Integer pageSize, Long userId);
    
    /**
     * 搜索问题
     */
    PageResult<QuestionDTO> searchQuestions(String keyword, Integer pageNum, Integer pageSize, Long userId);
    
    /**
     * 获取用户的问题列表
     */
    PageResult<QuestionDTO> getUserQuestions(Long authorId, Integer pageNum, Integer pageSize, Long userId);
    
    /**
     * 采纳答案
     */
    void acceptAnswer(Long questionId, Long answerId, Long userId);
    
    /**
     * 更新问题
     */
    Question updateQuestion(Long questionId, String title, String content, String tags, Long userId);
    
    /**
     * 删除问题
     */
    void deleteQuestion(Long questionId, Long userId);
}
